Enforce PSR2

dustinleblanc opened the pull request:
The defacto coding standard for PHP projects in the modern era is PSR2. Symfony Console which we are building upon is also PSR2. Terminus should follow this standard to encourage collaboration and easy contribution for the community. this PR won't effect our legacy code and makes this process incredibly simple, just run `robo sniff`. We already have Robo as a runtime dependency so we should start using it for development scripts rather than bash as well.

dustinleblanc committed:
Enforce PSR2

ari-gold commented:
@kimby77  Good one to keep on eye on for establishing our coding/design guidelines.

greg-1-anderson commented:
PSR-2 is a great idea; however, note that Terminus is [already running phpcs from Travis](https://github.com/pantheon-systems/terminus/blob/master/scripts/lint.sh#L5) to enforce the current Terminus coding conventions. Those references should become `--standard=PSR2` as well.

Also, I wouldn't just drop the RoboFile at the root of the project.  Put it in `scripts` with the other scripts.  Then, you can `require-dev` Robo and make a composer script to run sniff.  I can throw that change together if you have any question, and this PR is going to move forward.

dustinleblanc commented:
@greg-1-anderson Robo is currently a runtime dependency (dev-master too), should we move that our of require and into require-dev? I can move the file for sure, though I am not entirely sure of the purpose, is it simply for keeping the TLD smaller and easier to parse at a glance?

dustinleblanc committed:
Update dependencies

Upgrade to Symfony Console 3, Remove Robo as runtime dependency, add Robo as dev dependency.

dustinleblanc committed:
Import InputOption

dustinleblanc committed:
Move Robofile, linter to PSR2

dustinleblanc commented:
@greg-1-anderson the only part I'm not clear on is 

> make a composer script to run sniff

Can you clarify what to do, and why? Here or Slack is fine.

dustinleblanc commented:
@greg-1-anderson swapping out the  linter for PSR-2 makes Travis not so happy, I'll add an additional check to the test script for PSR-2 in the src dir only.

greg-1-anderson commented:
@dustinleblanc First step is to figure out whether the Terminus project is ready to move to PSR-2. As I mentioned in another issue, I think they should make a 0.x branch for the old composer; then, they could fix up composer.json, move code to src etc. on master. Messing with the test scripts to support some bifurcated system supporting different standards isn't a good idea. Let's wait for the go-ahead; then phpcbf can fix up all the violations (or at least most of them; I recommend validating with `phpcs -n --standard=PSR2` to keep this easy).  I'd be happy to demonstrate how to use Composer + Robo to make Robo-based build scripts in Terminus once we're ready to move in that direction.

dustinleblanc commented:
Sounds good, I'll be out the rest of the day anyway so I'm not in a hurry. I agree on the branch forking idea though, it'd make maintaining the two easier as well as making development on 1.0 more free and easy. Free and easy development ftw

SaraMcCutcheon commented:
I'm closing this because I took your advice and applied the PSR2 standard to the new src/ dir in #1167, where all 1.x work will be located. By the end of the project, the whole codebase will be PSR2-compliant. Thank you, @dustinleblanc!
